src/mod/MEs_Merge_FN.c

```diff
@@ -55,7 +56,20 @@ MEdge_ptr MEs_Merge_FN(MEdge_ptr e1, MEdge_ptr e2, int topoflag) {
     List_Delete(efaces2);
   }
 
-  ME_Delete(e2,0);
+  nsets = MESH_Num_MSets(mesh);
```

Comment:
The reason I opted to do it on the lowest level is to make sure that it's always done whenever the merging happens. If it's done on the higher level, and there is a new function added other than ME_Collapse that calls merge functions, than whoever writes it should remember to explicitly take care of this. 
Merging should be a relatively rare operation, so I thought trading a bit of performance for robustness with respect to code changes might be a good idea.